packages/cli/src/WorkflowRunner.ts

```diff
@@ -533,8 +532,8 @@ export class WorkflowRunner {
 				try {
 					// Check if this execution data has to be removed from database
 					// based on workflow settings.
-					let saveDataErrorExecution = config.get('executions.saveDataOnError') as string;
-					let saveDataSuccessExecution = config.get('executions.saveDataOnSuccess') as string;
+					let saveDataErrorExecution = config.getEnv('executions.saveDataOnError') as string;
+					let saveDataSuccessExecution = config.getEnv('executions.saveDataOnSuccess') as string;
```

Comment:
type castings?

Reply:
Removing these two assertions would require disentangling the `IWorkflowSettings` type, which would not be trivial is likely better tackled in a separate PR.